sys/fuchsia/fidl_mappings.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package fuchsia
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// layer indicates at which layer a FidlLibrary lives in the Fuchsia build
+// system.
+type layer int
+
+const (
+	_ layer = iota
+	zircon
+	garnet
+)
+
+// FidlLibrary describes a FIDL library. It captures required details such as
+// build location, header generation, etc.
+type FidlLibrary struct {
+	layer
+
+	// FqName stores the fully-qualified name of the library in parts, e.g.
+	// the `fuchsia.mem` library is `fuchsia`, `mem`.
+	FqName []string
+}
+
+// AllFidlLibraries lists all FIDL libraries.
+var AllFidlLibraries []FidlLibrary
+
+func init() {
+	add := func(layer layer, parts ...string) {
+		AllFidlLibraries = append(AllFidlLibraries, FidlLibrary{
+			layer:  layer,
+			FqName: parts,
+		})
+	}
+	add(zircon, "fuchsia", "mem")
```

Comment:
That construction was more complicated before, hence that `init` pattern. Switched it to a simple inline array literal, with unnamed field member for the struct literal since the definition is small. Let me know if you prefer named fields here.